nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/cosmos/document/AbstractAzureCosmosDBProcessor.java

```diff
@@ -0,0 +1,324 @@
+
+            switch(selectedConsistency) {
+                case CONSISTENCY_STRONG:
+                    clevel =  ConsistencyLevel.STRONG;
+                    break;
+                case CONSISTENCY_CONSISTENT_PREFIX:
+                    clevel = ConsistencyLevel.CONSISTENT_PREFIX;
+                    break;
+                case CONSISTENCY_SESSION:
+                    clevel = ConsistencyLevel.SESSION;
+                    break;
+                case CONSISTENCY_BOUNDED_STALENESS:
+                    clevel = ConsistencyLevel.BOUNDED_STALENESS;
+                    break;
+                case CONSISTENCY_EVENTUAL:
+                    clevel = ConsistencyLevel.EVENTUAL;
+                    break;
+                default:
+                    clevel = ConsistencyLevel.SESSION;
+            }
+            if (cosmosClient != null) {
+                closeClient();
+            }
+            getLogger().debug("Creating CosmosClient");
+            createDocumentClient(uri, accessKey, clevel);
+        }
+        getCosmosDocumentContainer(context);
+    }
+
+    protected void createDocumentClient(final String uri, final String accessKey, final ConsistencyLevel clevel) throws Exception {
+        try {
+            ConnectionPolicy connectionPolicy = ConnectionPolicy.getDefaultPolicy();
+            this.cosmosClient = new CosmosClientBuilder()
+                .setEndpoint(uri)
+                .setKey(accessKey)
+                .setConnectionPolicy(connectionPolicy).setConsistencyLevel(clevel)
+                .buildClient();
+        } catch (Exception e) {
+            getLogger().error("Failed to build cosmosClient {} due to {}", new Object[] { this.getClass().getName(), e }, e);
+            throw e;
+        }
+    }
+
+    protected abstract void doPostActionOnSchedule(final ProcessContext context);
+
+    protected void getCosmosDocumentContainer(final ProcessContext context) throws CosmosClientException {
+        final String databaseName = context.getProperty(DATABASE_NAME).getValue();
+        final String containerID = context.getProperty(CONTAINER_ID).getValue();
+        final String partitionKey = context.getProperty(PARTITION_KEY).getValue();
+
+        CosmosDatabase database = this.cosmosClient.createDatabaseIfNotExists(databaseName).getDatabase();
+
+        CosmosContainerProperties containerProperties =
+            new CosmosContainerProperties(containerID, "/"+partitionKey);
+
+        //  Create container with 400 RU/s by default if Not exists
+        this.container = database.createContainerIfNotExists(containerProperties, 400).getContainer();
+        doPostActionOnSchedule(context);
+    }
+
+    @OnStopped
+    public final void closeClient() {
+        if (connectionService == null && cosmosClient != null) {
+            // close client only when cosmoclient is created in Processor.
+            getLogger().debug("Closing CosmosClient");
```

Comment:
Can you wrap this in `logger.isDebugEnabled()`.

Reply:
wrapped as u suggested